### PR TITLE
Updated package with Null safety enabled.

### LINKED ISSUE #3: Regarding further maintainence of the package
Will this package be maintained further? I am working on a project that requires usage of protoo-client with latest features available. For now we are using https://github.com/ZackMitkin/dart-protoo-client, a fork of this repository, but soon we will need to publish our package so we won't be able to use it directly from github.

### PR DESCRIPTION
**Pull request by ritvij14**
- Updated package from https://github.com/ZackMitkin/dart-protoo-client
- Added Null Safety
- Updated version on `pubspec.yaml` and `CHANGELOG.md`

Resolves #3

### CODE REVIEW

**Comment by ritvij14** [comment]
@ZackMitkin what more things should I add to the README? Also, the description for this new version in CHANGELOG is ok or should I add something for it? (I changed the version number to 0.3.1, do tell if I should revert that)

File: lib/src/transports/NativeTransport.dart [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+      _logger.warn('send() failed:$error');
+    }
+  }
+
+  _onOpen() {
+    _logger.debug('onOpen');
+    this.safeEmit('open');
+  }
+
+  // _onClose(event) {
```
Comment (ZackMitkin) [inline lib/src/transports/NativeTransport.dart:56]: Are these comments still needed? Could we remove if the logic is already implemented?

File: test/proto_client_test.dart [hunk 2]
```diff
@@ -31,11 +29,11 @@ main() async {
 
   peer.on('request', (request, accept, reject) {
     print('request: ' + request.toString());
-    accept({ 'key1':"value1", 'key2':"value2"});
+    accept({'key1': "value1", 'key2': "value2"});
     //reject(404, 'Oh no~~~~~');
   });
 
-  await peer.connect();
+  //await peer.connect();
```
Comment (ZackMitkin) [inline test/proto_client_test.dart:36]: Why is this commented out? Does the test run?

**Review by ZackMitkin — COMMENTED** [review]
Looks really good. Great work! Just a few nitpicky things about comments and should be good.

File: test/proto_client_test.dart (hunk 2 above)
Comment (ritvij14) [reply test/proto_client_test.dart:36]: Oh my bad I was playing around with the code and might have commented it out and forgot to change it back by mistake.

Comment (ritvij14) [reply test/proto_client_test.dart:36]: image

Looks like i got mixed up in what I commented, this was done by you so I am not sure about the reason 😅

File: lib/src/transports/NativeTransport.dart (hunk 1 above)
Comment (ritvij14) [reply lib/src/transports/NativeTransport.dart:56]: I am unsure about this part, this complete function was commented out but I didn't exactly understand myself why it was.

File: test/proto_client_test.dart (hunk 2 above)
Comment (ZackMitkin) [reply test/proto_client_test.dart:36]: ah lol. Okay, yeah the PR looks good. It's usable, code quality shuldn't really be a priority here anyway. If someone is willing enough to cleanup and optimize so be it. But this should be good enough for pub.dev

**Pull request merged**